screens/Profile/ExternalProfileScreen/ExternalProfileScreen.tsx

```diff
@@ -1,55 +1,77 @@
-import { View, Text, TouchableOpacity } from "react-native"
+import React, { View, Text, TouchableOpacity } from "react-native"
 import { styles } from "./styles"
 import { globalStyles } from "../../../assets/global/globalStyles"
 import GeneralProfile from "../../../components/GeneralProfile/GeneralProfile"
 import ExpandableDescription from "../../../components/ExpandableDescription/ExpandableDescription"
 import SectionTabs from "../../../components/SectionTabs/SectionTabs"
-import { useState } from "react"
+import { useEffect, useState } from "react"
 import { profileStyles } from "../profileStyles"
 import { ProfileEvents } from "../ProfileEvents/ProfileEvents"
 import { ProfileInterests } from "../ProfileInterests/ProfileInterests"
 import { ProfileNetwork } from "../ProfileNetwork/ProfileNetwork"
+import { RouteProp, useRoute } from "@react-navigation/native"
+import { getAuth } from "firebase/auth"
+import LoadingScreen from "../../Loading/LoadingScreen"
+import { getUserData } from "../../../firebase/User"
+import { User } from "../../../types/User"
 
-type Contact = {
-  uid: string
-  firstName: string
-  lastName: string
-  profilePictureUrl?: string
-  description: string
-  interests: string[]
-  qualifications: string[],
-  location: string
-}
-
-const dummyProfile: Contact = {
-  uid: "4",
-  firstName: "Hervé",
-  lastName: "DelaStrite",
-  profilePictureUrl: "",
-  description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Dolor morbi non arcu risus quis. Mattis molestie a iaculis at. Tristique risus nec feugiat in fermentum posuere urna. Quisque sagittis purus sit amet volutpat consequat mauris nunc. Sapien eget mi proin sed libero. Condimentum mattis pellentesque id nibh tortor id aliquet lectus proin. Suspendisse faucibus interdum posuere lorem. In mollis nunc sed id semper risus. Amet consectetur adipiscing elit ut aliquam purus. Integer enim neque volutpat ac.",
-  interests: ["running"],
-  qualifications: ["bowling"],
-  location: "EPFL Ecublens"
-}
-
-/*
-* These lines are commented to not have an error with eslint, uncomment them to get the uid of the profile screen
-*/
-/*
 type RootStackParamList = {
   ExternalProfile: {
       uid: string;
-  };
-};
+  }
+}
 
-type ExternalProfileScreenRouteProp = RouteProp<RootStackParamList, 'ExternalProfile'>;
-*/
+type ExternalProfileScreenRouteProp = RouteProp<RootStackParamList, "ExternalProfile">
 
 const ExternalProfileScreen = () => {
+  const { uid } = useRoute<ExternalProfileScreenRouteProp>().params
```

Comment:
Should rename the variable to something like externalUserId, because it is a bit confusing with the 2 uids being present in this screen.